core/include/kernel/interrupt.h

```diff
@@ -87,4 +113,17 @@ void itr_set_affinity(size_t it, uint8_t cpu_mask);
  */
 void itr_core_handler(void);
 
+static inline void itr_add(struct itr_handler *handler, uint32_t type)
```

Comment:
We could pass the default type `IRQ_TYPE_NONE` also with this function. This would avoid many changes in this patch.
Opinions?

Reply:
I agree.
Ditto for `itr_alloc_add()`: could pass `IRQ_TYPE_NONE`.  One needing to allocate/add an interrupt with more parameters should call `itr_alloc_add_prio()` straight.

By the way, maybe rename `itr_add_prio()` to `itr_add_type_prio()`
and `itr_alloc_add_prio()` to `itr_alloc_add_type_prio()`.

Reply:
Ok, I'll do that. so in the end `itr_add` stay the same and `itr_add_type_prio` uses `type` and `prio` parameters.